galgebra/mv.py

```diff
@@ -1662,13 +1662,8 @@ def __mul__(self, sdopr):
                 terms += Dsdopl
             product = Sdop(terms, ga=sdopl.Ga)
             return Sdop.consolidate_coefs(product)
-        else:
-            if not isinstance(sdopl, Sdop):  # sdopl is a scalar
-                terms = [(sdopl * x[0], x[1]) for x in sdopr.terms]
-                product = Sdop(terms, ga=sdopr.Ga)  # returns Sdop
-                return Sdop.consolidate_coefs(product)
-            else:  # sdopr is a scalar or a multivector
-                return sum([x[0] * x[1](sdopr) for x in sdopl.terms], S(0))  # returns scalar
+        else:  # sdopr is a scalar or a multivector
+            return sum([x[0] * x[1](sdopr) for x in sdopl.terms], S(0))  # returns scalar
```

Comment:
Could use destruction here.

Reply:
Already working on that locally - I have a PR that I'll push out after this

Reply:
OK.